src/Common/CustomTagSelector/Types.ts

```diff
@@ -80,9 +80,7 @@ export interface ResizableTagTextAreaProps
     maxHeight?: number
     value: string
     refVar?: MutableRefObject<HTMLTextAreaElement>
-    dependentRef?:
-        | MutableRefObject<HTMLTextAreaElement>
-        | Record<string | number, MutableRefObject<HTMLTextAreaElement>>
-    dataTestId?: string
+    dependentRef?: MutableRefObject<HTMLTextAreaElement>
+    dependentRefs?: Record<string | number, MutableRefObject<HTMLTextAreaElement>>
```

Comment:
yes since we are accepting object for both it's difficult to obtain data type to determine which is which
null checks have been handled inside the component